vendor.py

```diff
@@ -0,0 +1,50 @@
+#!/usr/bin/env python3
+
+import fileinput
+import os.path
+
+output_lines = [
+    '# pylint: disable=ansible-format-automatic-specification,raise-missing-from',
+    'from __future__ import absolute_import, division, print_function',
+    '__metaclass__ = type',
+]
+buffer_lines = []
+with fileinput.input() as f:
+    for line in f:
+        if os.path.basename(fileinput.filename()) == '__init__.py':
+            fileinput.nextfile()
+            continue
+        if fileinput.isfirstline():
+            output_lines.extend(['', ''])
+        line = line.rstrip()
+
+        if line == '    basestring':
+            line = '    base_string = basestring'
+        elif 'basestring' in line:
+            line = line.replace('basestring', 'base_string')
+
+        if line == '':
+            output_lines.extend(buffer_lines)
+            buffer_lines.clear()
+            if output_lines and output_lines[-1] != '':
+                output_lines.append(line)
+        elif line.strip().startswith('# type:') or line.startswith('from apypie') or line.startswith('from __future__') or line.startswith('__all__'):
+            continue
+        elif line == 'import requests':
+            output_lines.extend(['try:', '    import requests', 'except ImportError:', '    pass'])
+        elif line in ['try:', 'if TYPE_CHECKING:'] or buffer_lines:
+            buffer_lines.append(line)
+            if ('pass' in line or 'TYPE_CHECKING =' in line or 'from apypie' in line) and ('from typing' in buffer_lines[1] or 'from apypie' in buffer_lines[1]):
+                buffer_lines.clear()
+        else:
+            if line.startswith('class ') or line.startswith('import '):
+                if not output_lines[-1].startswith('import '):
+                    if output_lines[-1] != '':
+                        output_lines.append('')
+                    output_lines.append('')
+            output_lines.append(line)
+    if buffer_lines:
+        output_lines.extend(buffer_lines)
+        buffer_lines.clear()
+
+print("\n".join(output_lines))
```

Comment:
Would it be easier to maintain a stack of patches? i.e. with quilt?

Reply:
It's been a while I've seen "easier" and "quilt" in one sentence ;-)

I am not sure. The above should work rather reliably even if we do changes to apypie, a patch (or a series) might not. And something would still need to do the job of "concat all the files together" (well, `cat` exists, granted).

Reply:
Let's keep it that way, and time will show.